### DISCUSSION TITLE
material/mat-table-filter/

### DISCUSSION THREAD

**Discussion by giscus** [arungudelli/angularjswiki#179] [Announcements]
# material/mat-table-filter/

In this tutorial we will learn how to filter a mat-table in Angular.

https://www.angularjswiki.com/material/mat-table-filter/

**Comment by CMercs** [comment]
Why can't we use record[key]????  I had to use record[<string>key], but that seems more intuitive than having to hard Type it (I couldn't because I'm building it dynamically.)

Also, I'd like to include the option to have a standard input box search as well.   So for example; select column filters via dropdown AND have an input field that searches all fields for a string.  How can I do this at the same time.  FilterPredicate seems to expect different things for each type.

**Reply by arungudelli** [reply]
Ideally We can.. But typescript compiler restrict us to use `record[key]` directly.

**Reply by arungudelli** [reply]
As mentioned in https://www.angularjswiki.com/material/mat-table-filterpredicate/#what-is-mat-table-filterpredicate article.. You can write a function which uses default mat filter predicate and call it after the column filter.. 

Ideally after filtering by columns, convert the column values to data string(refer the article) and check whether the input search string exist in it or not.. Let me know if you face any issues..

**Reply by arungudelli** [reply]
@CMercs did u solve your problem?
